src/Block/MediaBlockService.php

```diff
@@ -209,21 +203,17 @@ protected function getFormatChoices(?MediaInterface $media = null): array
     protected function getMediaBuilder(FormMapper $form): FormBuilderInterface
     {
         // simulate an association ...
-        $fieldDescription = $this->getMediaAdmin()->getModelManager()->getNewFieldDescriptionInstance($this->mediaAdmin->getClass(), 'media', [
+        $fieldDescription = $this->fieldDescriptionFactory->create($this->mediaAdmin->getClass(), 'media', [
```

Comment:
```suggestion
        $fieldDescription = $this->mediaAdmin->createFieldDescription('media', [
```
There is a `AdminInterface::createFieldDescription()` method, so there is no need of injecting `FieldDescriptionFactory`.

Reply:
oh that's better

Reply:
This methods also calls the setAdmin, so I can remove that line

Reply:
Done.